backend/handler/database/roms_handler.py

```diff
@@ -477,17 +462,88 @@ def build_func(provider: str, column: InstrumentedAttribute):
 
             # Create a subquery that identifies the primary ROM in each group
             # Priority order: is_main_sibling (desc), then by fs_name_no_ext (asc)
+            base_subquery = query.subquery()
             group_subquery = (
-                session.query(Rom.id)
+                select(base_subquery.c.id)
+                .select_from(base_subquery)
+                .with_only_columns(
+                    base_subquery.c.id,
+                    base_subquery.c.fs_name_no_ext,
+                    base_subquery.c.platform_id,
+                    base_subquery.c.igdb_id,
+                    base_subquery.c.ss_id,
+                    base_subquery.c.moby_id,
+                    base_subquery.c.ra_id,
+                    base_subquery.c.hasheous_id,
+                    base_subquery.c.launchbox_id,
+                    base_subquery.c.tgdb_id,
+                )
```

Comment:
@zurdi15 here we only fetch the colums we need to build the subquery, saving us from loading metadata JSON for every game in the filtered selection (which for homepage is all games)